src/components/timeout-dialog/timeout-dialog.browser.test.js

```diff
@@ -0,0 +1,408 @@
+    await clockTickSeconds(page, 799);
+    await expect(page).not.toShowTimeoutDialog();
+    await clockTickSeconds(page, 1);
+    await expect(page).toShowTimeoutDialog();
+  });
+
+  it('should time you out if you reach the timeout after having previously extended your session', async () => {
+    await renderTimeoutDialog(page, `
+      data-timeout="900"
+      data-countdown="100"
+      data-keep-alive-url="?keepalive"
+      data-sign-out-url="/timeout-reached"
+    `);
+    await clockTickSeconds(page, 800);
+    await expect(page).toClick('button', { text: 'Stay signed in' });
+    await clockTickSeconds(page, 900);
+    await page.waitForNavigation({ timeout: 500 });
+    await expect(page.url()).toMatch('/timeout-reached');
+  });
+
+  it('should let you extend your session repeatedly', async () => {
+    await renderTimeoutDialog(page, `
+      data-timeout="900"
+      data-countdown="100"
+      data-keep-alive-url="?keepalive"
+      data-sign-out-url="/timeout-reached"
+    `);
+    await clockTickSeconds(page, 800);
+    await expect(page).toClick('button', { text: 'Stay signed in' });
+    await clockTickSeconds(page, 800);
+    await expect(page).toClick('button', { text: 'Stay signed in' });
+    await clockTickSeconds(page, 100);
+    await delay(500);
+    await expect(page.url()).not.toMatch('/timeout-reached');
+    await clockTickSeconds(page, 900);
+    await page.waitForNavigation({ timeout: 500 });
+    await expect(page.url()).toMatch('/timeout-reached');
+  });
+
+  it('should let you sign out early', async () => {
+    await renderTimeoutDialog(page, `
+      data-timeout="900"
+      data-countdown="100"
+      data-keep-alive-url="?keepalive"
+      data-sign-out-url="/timeout-reached"
+    `);
+    await clockTickSeconds(page, 800);
+    await expect(page).toClick('a', { text: 'Sign out' });
+    await page.waitForNavigation({ timeout: 500 });
+    await expect(page.url()).toMatch('/timeout-reached');
+  });
+
+  it('should let you specify a separate timeout url', async () => {
+    await renderTimeoutDialog(page, `
+      data-timeout="900"
+      data-countdown="100"
+      data-keep-alive-url="?keepalive"
+      data-timeout-url="/timeout-reached"
+      data-sign-out-url="/signed-out-early"
+    `);
+    await clockTickSeconds(page, 900);
+    await page.waitForNavigation({ timeout: 500 });
+    await expect(page.url()).toMatch('/timeout-reached');
+    await renderTimeoutDialog(page, `
+      data-timeout="900"
+      data-countdown="100"
+      data-keep-alive-url="?keepalive"
+      data-timeout-url="/timeout-reached"
+      data-sign-out-url="/signed-out-early"
+    `);
+    await clockTickSeconds(page, 800);
+    await expect(page).toClick('a', { text: 'Sign out' });
+    await page.waitForNavigation({ timeout: 500 });
+    await expect(page.url()).toMatch('/signed-out-early');
+  });
+
+  // TODO at the moment it doesn't, should we change test or implementation?
```

Comment:
This is an interesting, worth capturing somewhere?

Reply:
yup, could probably be a tech debt ticket, I'll create one and link it here